### PR TITLE
support verification for noop and exists for v3 circuit

### PR DESCRIPTION
**Pull request by vmidyllic**

### CODE REVIEW

File: src/proof/verifiers/query.ts [hunk 1]
```diff
@@ -221,15 +205,46 @@ async function validateDisclosure(
...
     ldLoader
   );
-  if (bi !== outputs.value[0]) {
-    throw new Error(`value that was used is not equal to requested in query`);
+  if (bi !== outputs.operatorOutput) {
+    throw new Error(`operator output should be equal to disclosed value`);
+  }
+
+  if (outputs.operator !== Operators.SD) {
+    throw new Error(`operator for selective disclosure must be $eq`);
```
Comment (volodymyr-basiuk) [inline src/proof/verifiers/query.ts:226]: operator for native SD is SD

File: src/proof/verifiers/query.ts [hunk 2]
```diff
@@ -221,15 +205,46 @@ async function validateDisclosure(
...
+      throw new Error(`public signal values must be zero`);
+    }
+  }
+}
+export async function validateEmptyCredentialSubjectNoopNativeSupport(
+  cq: QueryMetadata,
+  outputs: ClaimOutputs
+) {
+  if (outputs.operator !== Operators.NOOP) {
+    throw new Error('empty credentialSubject request available only for equal operation');
```
Comment (volodymyr-basiuk) [inline src/proof/verifiers/query.ts:240]: only for noop (native)

**Review by volodymyr-basiuk — APPROVED** [review]

File: src/proof/verifiers/query.ts (hunk 2 above)
Comment (vmidyllic) [reply src/proof/verifiers/query.ts:240]: fixed

File: src/proof/verifiers/query.ts (hunk 1 above)
Comment (vmidyllic) [reply src/proof/verifiers/query.ts:226]: fixed

**Review by volodymyr-basiuk — DISMISSED** [review]

File: src/proof/verifiers/query.ts [hunk 3]
```diff
@@ -221,15 +205,46 @@ async function validateDisclosure(
...
+) {
+  if (outputs.operator !== Operators.NOOP) {
+    throw new Error('empty credentialSubject request available only for $noop operation');
+  }
+  for (let index = 1; index < outputs.value.length; index++) {
+    if (outputs.value[index] !== 0n) {
+      throw new Error(`empty credentialSubject request not available for array of values`);
+    }
+  }
+  return;
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply src/proof/verifiers/query.ts:247]: fixed

File: src/proof/verifiers/query.ts [hunk 4]
```diff
@@ -221,15 +205,46 @@ async function validateDisclosure(
...
+  }
+
+  for (let index = 0; index < outputs.value.length; index++) {
+    if (outputs.value[index] !== 0n) {
+      throw new Error(`public signal values must be zero`);
+    }
+  }
+}
+export async function validateEmptyCredentialSubjectNoopNativeSupport(
+  cq: QueryMetadata,
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply src/proof/verifiers/query.ts:236]: fixed

File: src/proof/provers/inputs-generator.ts [hunk 5]
```diff
@@ -237,6 +238,8 @@ export class InputGenerator {
     circuitInputs.profileNonce = BigInt(params.authProfileNonce);
     circuitInputs.skipClaimRevocationCheck = params.skipRevocation;
```
[the remark this answers is not in the document]
Comment (vmidyllic) [reply src/proof/provers/inputs-generator.ts:240]: fixed

**Review by volodymyr-basiuk — APPROVED** [review]

**Pull request merged**